services/QuillLMS/client/app/bundles/Grammar/components/admin/admin.tsx

```diff
@@ -39,7 +39,12 @@ class AdminContainer extends React.Component<AdminContainerProps> {
       }
     }
     );
-    this.props.dispatch(questionsActions.startListeningToQuestions());
+    function getUIDFromURL(url: string): string {
+      const parts = url.split('/');
+      return parts[parts.length - 2]
+    }
+    const activityUID = getUIDFromURL(this.props.location.pathname)
+    this.props.dispatch(questionsActions.startListeningToQuestions(activityUID));
```

Comment:
Does this component normally have an activity uid parameter? It looks to me like this is the top-level parent component for the whole admin interface, so I wonder if we want to keep this questions call generic so we have them all in the reducer for the `/grammar/#/admin/questions` page and other places we might actually want all of the questions.

Reply:
you're right, it seems the grammar admin interface is set up so that it expects to have the list of *all* the grammar questions. That's probably why the student-facing one does the same.